expression/builtin_control.go

```diff
@@ -224,8 +224,8 @@ func (c *caseWhenFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	}
 	// Set retType to BINARY(0) if all arguments are of type NULL.
 	if fieldTp.GetType() == mysql.TypeNull {
-		fieldTp.SetFlen(0)
-		fieldTp.SetDecimal(types.UnspecifiedLength)
+		fieldTp.SetFlen(1)
+		fieldTp.SetDecimal(0)
```

Comment:
Looks like you set both flen and decimal to 0 [here](https://github.com/pingcap/tidb/pull/34399/files#diff-ec0d510c2381cfee88c8e64a22928d504b6dbf0a9ce42b93cfca1a279653d1c4R156) for null type.

Reply:
change to 0,0.